app/assets/javascripts/shared/dradis_datatable.js

```diff
@@ -20,6 +22,18 @@ class DradisDatatable {
       }
     });
 
+    // Only show default columns on first load
+    var hiddenColumnIndexes = [];
+    if (localStorage.getItem(this.localStorageKey) === null) {
+      this.tableHeaders.forEach(function(column, index) {
+        var columnName = column.textContent.trim();
+
+        if (!that.defaultColumns.includes(columnName)) {
+          hiddenColumnIndexes.push(index);
+        }
+      });
+    }
```

Comment:
Same anti pattern:

1. comment starts talking about the columns you will *show*
2. variable name flips that to track the *hidden* columns
3. when there's no local storage you loop through a list of columns you want to *show* (the default list)
4. but check that the current column is *not* on that list, and then you add it to the list you want to *hide*

Since what you're passing as "default columns" is the show list, and what you're storing in local storage is the "show list" too, I think this code can be made clearer.

Reply:
Yeah it is a bit weird, but it is done this way because there is no way to tell datatables to show only specific columns as all columns are shown by default. So I had to do it reversely by hiding certain columns, instead of _only_ showing certain columns.

The data in local storage is handled by datatables itself. So the flow currently is:
1. Script checks if local storage key exists
1.a) If yes, datatables load the previous state

2. If no:
2.a) script specifies columns to hide
2.b) Datatables automatically save the state to local storage

 An example of the data that datatables saves to local storage is shown below. How they hide/show columns is by passing in the `visible: true` or `visible: false` property to the elements in the columns array.

```javascript
{
    "columns": [
        {
            "search": {
                "caseInsensitive": true,
                "regex": false,
                "search": "",
                "smart": true
            },
            "visible": true
        },
        {
            "search": {
                "caseInsensitive": true,
                "regex": false,
                "search": "",
                "smart": true
            },
            "visible": true
        },
        {
            "search": {
                "caseInsensitive": true,
                "regex": false,
                "search": "",
                "smart": true
            },
            "visible": true
        },
        {
            "search": {
                "caseInsensitive": true,
                "regex": false,
                "search": "",
                "smart": true
            },
            "visible": true
        },
        {
            "search": {
                "caseInsensitive": true,
                "regex": false,
                "search": "",
                "smart": true
            },
            "visible": false
        },
        {
            "search": {
                "caseInsensitive": true,
                "regex": false,
                "search": "",
                "smart": true
            },
            "visible": false
        },
        {
            "search": {
                "caseInsensitive": true,
                "regex": false,
                "search": "",
                "smart": true
            },
            "visible": true
        },
        {
            "search": {
                "caseInsensitive": true,
                "regex": false,
                "search": "",
                "smart": true
            },
            "visible": true
        },
        {
            "search": {
                "caseInsensitive": true,
                "regex": false,
                "search": "",
                "smart": true
            },
            "visible": true
        },
        {
            "search": {
                "caseInsensitive": true,
                "regex": false,
                "search": "",
                "smart": true
            },
            "visible": true
        }
    ],
    "length": 25,
    "order": [
        [
            0,
            "asc"
        ]
    ],
    "search": {
        "caseInsensitive": true,
        "regex": false,
        "search": "",
        "smart": true
    },
    "start": 0,
    "time": 1621489207448
}
```